core/transaction.go

```diff
@@ -21,6 +21,47 @@ import (
 	"golang.org/x/crypto/sha3"
 )
 
+type Resource uint32
+
+const (
+	ResourceL1Gas Resource = iota + 1
+	ResourceL2Gas
+)
+
+func (r Resource) String() string {
+	switch r {
+	case ResourceL1Gas:
+		return "L1_GAS"
+	case ResourceL2Gas:
+		return "L2_GAS"
+	default:
+		return ""
+	}
+}
+
+type DataAvailabilityMode uint32
+
+const (
+	DAModeL1 DataAvailabilityMode = iota
+	DAModeL2
+)
+
+type FeeUnit byte
+
+// WEI is the default value since ETH is the first native currency.
+// This allows us to avoid a costly database migration.
+const (
+	WEI FeeUnit = iota
+	STRK
+)
+
+// From the RPC spec: The max amount and max price per unit of gas used in this transaction.
+type ResourceBounds struct {
+	MaxAmount uint64
+	// MaxPricePerUnit is technically a uint128
+	MaxPricePerUnit *felt.Felt
```

Comment:
Something I was wondering while implementing: Should we have a `uint128` type? Might save a little disk space.

Reply:
Yeah, could be worthwhile to explore later on.